### PR TITLE
feat: add `run-dash` action to analyze 3rd party dependencies

### LINKED ISSUE #168: [Workflows]: Add a GitHub action, that downloads and runs `dash-licenses`
## Description

We want to further automate the IP due diligence for our project. To achieve that, we should create a GitHub action,
that installs and runs the Eclipse [dash-licenses](https://github.com/eclipse/dash-licenses) tool.

The action should download the latest version of `dash-licenses` and install all its prerequisites.
It should then generate a temporary `DEPENDENCIES` file, like all of our repositories have included. 
The action can then give hints on:

1) Is the current `DEPENDENCIES` file up-to-date
2) Are there `restricted` libraries included, that we need to create IP issues for (still manual step)
3) Are there even `rejected` dependencies, that we have to remove

As `input` the action could receive the `dash-licenses` input file, which is specific to every build tool. Additionally, the path to the `DEPENDENCIES` file, the up-to-date comparison is made against could be specified. This will allow repos with multiple `DEPENDENCIES` (i.e. separate frontend and backend) to use the same action.

### PR DESCRIPTION
**Pull request by SebastianBezold**
## Description

This PR does add a new GitHub action, that is setting up the [Eclipse Dash](https://github.com/eclipse/dash-licenses) license tool. Based on a view config params, Dash will analyze the 3rd party dependencies and check, if the `DEPENDENCIES` file present in a repository is up-to-date.

Fixes #168

### CODE REVIEW

**Review by almadigabor — CHANGES_REQUESTED** [review]

**Review by FaGru3n — APPROVED** [review]
Great work looks easy to understand an will automate the Dash Licence Check 


suggested changes:
- i would only add an empty line in .github/actions/run-dash/action.yaml at the end

File: .github/actions/run-dash/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+`restricted` or `rejected` libraries. The results are provided as boolean action outputs.
+
+## Usage
+
+The following examples show different usage of the action for different build systems and programming languages.
+
+### Analyze golang application
+
+Golang defines detailed dependency and version info in a file called `go.sum`. The Eclipse Dash tool can interpret that file
+as-is. Therefor a minimal set of config for the `run-dash` action is needed.
```
[the remark this answers is not in the document]
Comment (SebastianBezold) [reply .github/actions/run-dash/README.md]: Oh thx for pointing out :P fixed that

**Review by almadigabor — APPROVED** [review]
Way better to inform what's wrong with the file. I was wondering if it would be more useful to output a combined result like it's out of date & contains rejected & contains restricted.

**Comment by SebastianBezold** [comment]
> Way better to inform what's wrong with the file. I was wondering if it would be more useful to output a combined result like it's out of date & contains rejected & contains restricted.

You actually could do both. Via input, you can configure when the action itself will fail, but the flags for that (out-of-date, contains restricted, contains rejected) are also available as output.
So you could configure it to never fail and evaluate the step output yourself

**Pull request merged**